src/teleop/gui/src/components/JointStateTable.vue

```diff
@@ -0,0 +1,180 @@
+                <col style="width: 60px">
+                <col style="width: 75px">
+                <col style="width: 85px">
+            </colgroup>
+
+            <thead>
+
+                <tr class="Bold">
+                    <th class="tableElement">Motor</th>
+                    <th class="tableElement">Positon (m)</th>
+                    <th class="tableElement">Velocity (m/s)</th>
+                    <th class="tableElement">Effort (Nm)</th>
+
+                </tr>
+            </thead>
+
+            <tbody>
+
+                <tr v-for="motor in motors">
+                    <th class="tableElement">{{motor.name}}</th>
+                    <td class="tableElement">{{motor.position}} </td>
+                    <td class="tableElement">{{motor.velocity}} </td>
+                    <td class="tableElement">{{motor.effort}} </td>
+
+                </tr>
+            </tbody>
+        </table>
+    </div>
+
+    <div v-else>
+        <div>
+            <h3> Motor Data </h3>
+        </div>
+        <table class="tableFormat" style="undefined;table-layout: fixed; width: 745px">
+            <colgroup>
+                <col style="width: 85px">
+                <col style="width: 60px">
+                <col style="width: 75px">
+                <col style="width: 85px">
+                <col style="width: 60px">
+                <col style="width: 75px">
+                <col style="width: 85px">
+                <col style="width: 60px">
+                <col style="width: 75px">
+                <col style="width: 85px">
+            </colgroup>
+
+            <thead>
+
+                <tr class="Bold">
+                    <th class="tableElement">Motor</th>
+                    <th v-for="motor in motors" class="tableElement">{{motor.name}}</th>
+                </tr>
+                <tr class="Bold">
+                    <th class="tableElement">Position (m)</th>
+                    <td v-for="motor in motors" class="tableElement">{{motor.position}} </td>
+                </tr>
+                <tr class="Bold">
+                    <th class="tableElement">Velocity (m/s)</th>
+                    <td v-for="motor in motors" class="tableElement">{{motor.velocity}} </td>
+                </tr>
+                <tr class="Bold">
+                    <th class="tableElement">Effort (Nm)</th>
+                    <td v-for="motor in motors" class="tableElement">{{motor.effort}} </td>
+                </tr>
+            </thead>
+
+        </table>
+
+    </div>
+</div>
+</template>
+
+<script>
+import ROSLIB from "roslib"
+
+export default {
+    data() {
+        return {
+            motors: []
+
+        }
+    },
+    props: {
+        JointStateData: {
+            type: Object,
+            required: true
+        },
+
+        vertical: {
+            type: Boolean,
+            required: true
+        }
+    },
+
+    watch: {
+        JointStateData: function (JointStateData) {
```

Comment:
Probably worth making a comment noting what we do here. While it is technically not neccesary to reorganize the data like this, it makes it much cleaner for the v-for loop in the html section. Also please switch this to camel case.